syntax = "proto3";
package m2.pb;

import "m2g_SpriteType.proto";
import "m2g_AnimationStateType.proto";
import "m2g_AnimationType.proto";

message AnimationState {
  m2g.pb.AnimationStateType type = 1;
  repeated m2g.pb.SpriteType sprites = 2;
}

message Animation {
  m2g.pb.AnimationType type = 1;
  float fps = 2;
  repeated AnimationState states = 3;
  m2g.pb.AnimationStateType initial_state = 4;
}

message Animations {
  repeated Animation animations = 1;
}
